### DISCUSSION TITLE
MeanAveragePrecision labels format for update method

### DISCUSSION THREAD

**Discussion by suchith-sixsense** [Lightning-AI/torchmetrics#1253] [CompVision] [answered]
Hello all,

I am new to torch-metrics and have been using the library for computing `mAP`. I have python 3.6.8 installed in one of the server and there I could able to install `torchmetrics == 0.8.2` (which is not the latest version when writing the post). As I was going through the docs [here](https://torchmetrics.readthedocs.io/en/v0.8.2/detection/mean_average_precision.html) of `v0.8.2`, we have to use update method where it takes `preds` and `targets`. The docs is given below

![image](https://user-images.githubusercontent.com/110653781/194144518-cac3319f-8384-40c4-8b7a-b855ae7e007f.png)

1. What does it mean when prediction needs in `0-indexed detection classes` and ground-truth needs `1-indexed detection classes` ?

2. Note that in the latest version (`0.10.0`) the [doc](https://torchmetrics.readthedocs.io/en/stable/detection/mean_average_precision.html) says both ground-truth and prediction needs `0-indexed detection class`. What does this change mean ?

Also I checked that after version `0.8.2`, the change in docs appeared in [v0.9.1](https://torchmetrics.readthedocs.io/en/v0.9.1/detection/mean_average_precision.html)

**Comment by twsl** [comment, marked as answer]
I think it actually doesn't matter because the `class_ids` are never iterated by index and mAP will simply calculate the results for each class. The 1-indexed ground truth might stem from the fact, that many object detection models require the first index in the class labels to be the "nothing" or "background" label. However, the implementation evaluates all annotated detections.

**Reply by suchith-sixsense** [reply]
Yes, you are right like all that is important is ground truth class is same as prediction class. I was confused by the wording "0-indexed detection classes" meaning somehow the prediction `class_id + 1` is compared with ground truth `class_id` as both are different indexed.

As of now I understand that make sure the class_ids is same for both prediction and ground truth, as the implementations directly compares the `class_id`
